k8s/agent-scaler/agent-scaler.sh

```diff
@@ -99,17 +99,27 @@ while true; do
         # How many slots are available globally before hitting max_jobs
         slots_available_globally=$((max_jobs - num_all_active_joshua_jobs))
 
-        # Determine how many jobs this scaler instance will attempt to start in this cycle
-        num_to_attempt_this_cycle=${batch_size} # Start with batch_size as the base
+        # Determine how many jobs this scaler instance will attempt to start in this cycle.
+        # We want to be aggressive but also bounded.
+        # Option 1: How many jobs would we need to clear the queue in one go (using batch_size)?
+        num_to_clear_queue=$(( (num_ensembles + batch_size - 1) / batch_size ))
```

Comment:
This is confusing to me. `num_to_clear_queue` seems to mean how many batches will be needed.
However, below we have `num_to_clear_queue < slots_available_globally` comparison, where `slots_available_globally` is the number of jobs, thus comparing batches to jobs?